packages/typescript-estree/src/convert.ts

```diff
@@ -374,51 +377,45 @@ export class Converter {
       type: customType,
     });
 
-    Object.keys(node)
+    if ('type' in node) {
+      result.typeAnnotation =
+        node.type && 'kind' in node.type && ts.isTypeNode(node.type)
```

Comment:
in case if we receive JSDoc nodes type may be of invalid type, `JSDocReturnTag` or `SyntheticExpression`.

`deeplyCopy` is used for unknown/unsupported nodes and hits from supported typescript versions should === 0, unless we are missing support for nodes.